src/publisher/views/publish/long-build.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+
+import Layout from '../components/Layout';
+import DatasetStatus from '../../../shared/views/components/dataset/DatasetStatus';
+import T from '../../../shared/views/components/T';
+import { CubeBuildStatus } from '../../../shared/enums/cube-build-status';
+
+export default function LongBuild(props) {
+  const spinner = {
+    display: 'inline-block',
+    height: '50px',
+    width: '50px',
+    verticalAlign: 'middle',
+    marginLeft: '50px'
+  };
+  const showSpinner = () => {
```

Comment:
When using react components, it's better to make them standalone rather than nesting them inside other components and relying on parent props. It's cleaner and makes it easier to share them with other templates in future. It's also better for performance reasons when the templates are rendered client-side (not really an issue here) so it's a good habit in general.

e.g. move this outside of the parent component `LongBuild`:

```tsx
function ShowSpinner({ buildLogEntry }) {
  ...
}
```

and call it like 

```tsx
   <ShowSpinner buildLogEntry={buildLogEntry} />
```

or just pass all the props if you're using lots of stuff.

```tsx
  <ShowSpinner props={...props} />
```